File: Jonghyeok Kim/2022-07-18-Merge-Intervals.py
from typing import List

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=(lambda x: x[0]))
        output = [intervals[0]]
        for start, end in intervals:
            if start <= output[-1][1]:
                output[-1][1] = max(end, output[-1][1])
            else:
                output.append([start,end])
        return output

File: Chloe Gwon/2022-07-18-Merge-Intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    const len = intervals.length;
    if (len<2) return intervals;
    
    intervals.sort((a,b) => a[0] - b[0]);
    const res = [intervals[0]];
    let resIdx = 0;
    
    for (let i=1; i<len; i++){
        if (res[resIdx][1] >= intervals[resIdx][i]){
            res[resIdx][1] = Math.max(intervals[i][1], res[resIdx][1]);
        }else{
            res.push(intervals[i]);
            resIdx++;
        }
    }
    return res;
};
